packages/migrate/src/__tests__/DbExecute.test.ts

```diff
@@ -0,0 +1,702 @@
+    })
+
+    it('should pass using a transaction with --file --schema', async () => {
+      ctx.fixture('schema-only-sqlserver')
+
+      fs.writeFileSync(
+        'script.sql',
+        `-- start a transaction
+BEGIN TRANSACTION;
+
+SELECT 1
+      
+-- commit changes    
+COMMIT;`,
+      )
+      const result = DbExecute.new().parse([
+        '--preview-feature',
+        '--schema=./prisma/schema.prisma',
+        '--file=./script.sql',
+      ])
+      await expect(result).resolves.toMatchInlineSnapshot(`Script executed successfully.`)
+    })
+
+    // Limitation of sqlserver
+    // DROP DATABASE statement cannot be used inside a user transaction.
+    it('should fail if DROP DATABASE in a transaction with --file --schema', async () => {
+      ctx.fixture('schema-only-sqlserver')
+
+      fs.writeFileSync(
+        'script.sql',
+        `-- start a transaction
+BEGIN TRANSACTION;
+
+${sqlScript}
+      
+-- commit changes    
+COMMIT;`,
+      )
+      const result = DbExecute.new().parse([
+        '--preview-feature',
+        '--schema=./prisma/schema.prisma',
+        '--file=./script.sql',
+      ])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              DROP DATABASE statement cannot be used inside a user transaction.
+
+
+            `)
+    })
+
+    it('should fail with P1013 error with invalid url with --file --url', async () => {
+      ctx.fixture('schema-only-sqlserver')
+      expect.assertions(2)
+
+      fs.writeFileSync('script.sql', '-- empty')
+      try {
+        await DbExecute.new().parse(['--preview-feature', '--url=invalidurl', '--file=./script.sql'])
+      } catch (e) {
+        expect(e.code).toEqual('P1013')
+        expect(e.message).toMatchInlineSnapshot(`
+          P1013
+
+          The provided database string is invalid. Error parsing connection string: relative URL without a base in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
+
+        `)
+      }
+    })
+
+    it('should fail with P1001 error with unreachable url with --file --url', async () => {
+      ctx.fixture('schema-only-sqlserver')
+      expect.assertions(2)
+
+      fs.writeFileSync('script.sql', '-- empty')
+      try {
+        await DbExecute.new().parse([
+          '--preview-feature',
+          '--url=sqlserver://doesnotexist:1433;database=tests-migrate;user=SA;password=Pr1sm4_Pr1sm4;trustServerCertificate=true;',
+          '--file=./script.sql',
+        ])
+      } catch (e) {
+        // TODO: Code not defined?
+        expect(e.code).toEqual(undefined)
+        expect(e.message).toMatchInlineSnapshot(`
+          Error creating a database connection.
+
+
+        `)
+      }
+    })
```

Comment:
SQL Server
Here it's interesting to see that there is no error code, I was expecting `P1001` like in PostgreSQL or MySQL
@tomhoule any idea?

Reply:
https://github.com/prisma/prisma/issues/11407